### DISCUSSION TITLE
客户端使用 CDN 优选 IP 如何配置

### DISCUSSION THREAD

**Discussion by jaesony** [lxhao61/integrated-examples#224] [Q&A] [answered]
客户端xhttp tls参考配置里cdn address用的是域名，我在tlsSettings里加了serverName，用cloudflare优选ip一直连接不上。
还有请教下如采用xhttp reality，cdn又要怎么配置？

**Comment by lxhao61** [comment]
客户端 address 用的域名改为 IP（含优选 IP），tlsSettings 里加 serverName 配置 IP 对应的域名。

**Reply by jaesony** [reply]
"tlsSettings": {
          "serverName": "cdn.xxx.yyy",
          "alpn": ["h2"],
          "fingerprint": "chrome"
address用优选ip，客户端加了serverName，连不上。用cdn域名倒没问题

**Reply by lxhao61** [reply]
注意此域名不能启用 CDN。

**Reply by jaesony** [reply]
没明白您的意思。服务端用的Xray(M+H+K+A)+Caddy(N+T)，xhttp tls用的cdn域名。客户端用的client configuration里的xhttp tls分离配置。现在的问题就是xhttp分离时cdn用域名没问题，只是用不了优选ip

**Reply by lxhao61** [reply]
优选 IP 的域名是由 Cloudflare 解析的，还是？
优选 IP 的域名不能启用 CDN。

**Reply by jaesony** [reply]
是由cf解析的，也启用了小云朵。优选ip不能启用cdn是什么意思，那优选ip怎么关联域名？

**Reply by lxhao61** [reply, marked as answer]
晕！简单说使用优选 IP 注意事项：其对应域名必由 CDN 提供商解析，但不能启用 CDN。
